uploadNeuroDB/NeuroDB/objectBroker/MriUploadOB.pm

```diff
@@ -0,0 +1,184 @@
+          1, '/tmp/my_tarchive.tar.gz', 1
+      );
+  } catch(NeuroDB::objectBroker::ObjectBrokerException $e) {
+      die sprintf(
+          "Failed to retrieve MRI uploads: %s",
+          $e->errorMessage
+      );
+  }
+
+=head1 DESCRIPTION
+
+This class provides a set of methods to either fetch or insert mri upload
+records. The operations are always performed on database table C<mri_upload>.
+Each method will throw a C<NeuroDB::objectBroker::ObjectBrokerException> if 
+the request could not be performed successfully.
+
+=head2 Methods
+
+=cut
+
+use Moose;
+use MooseX::Privacy;
+
+use File::Basename;
+
+use NeuroDB::Database;
+use NeuroDB::DatabaseException;
+use NeuroDB::objectBroker::ObjectBrokerException;
+
+use TryCatch;
+
+# These are the only fields modified when inserting a new MRI upload record
+my @MRI_UPLOAD_FIELDS = ('UploadedBy' ,'UploadDate','TarchiveID','DecompressedLocation');
+
+=pod
+
+=head3 new(db => $db) >> (constructor)
+
+Create a new instance of this class. The only parameter to provide is the
+C<Database> object used to access the database.
+
+INPUT: the database object used to read/modify the C<mri_upload> table.
+
+RETURNS: new instance of this class.
+
+=cut
+
+has 'db'     => (is  => 'rw', isa => 'NeuroDB::Database', required => 1);
+
+=pod
+
+=head3 getWithTarchive($isCount, $tarchiveLocation, $isBaseNameMatch)
+
+Fetches the entries in the C<mri_upload> table that have a specific archive
+location. This method throws a C<NeuroDB::objectBroker::ObjectBrokerException>
+if the operation could not be completed successfully.
+
+INPUT: a boolean indicating if only a count of the records found is needed
+       of the full record properties
+       the path of the archive location
+       a boolean indicating if a match is sought on the full archive name
+       or only the basename
+
+RETURNS: a reference to an array of array references. If C<$isCount> is true, then
+         C<$returnValue->[0]->[0]> will contain the count of records sought. Otherwise
+         C<$returnValue->[x]->[y]> will contain the value of the yth column (in array
+         C<@MRI_UPLOAD_FIELDS> for the xth record retrieved.
+
+=cut
+
+sub getWithTarchive {
+    my($self, $isCount, $tarchiveLocation, $isBaseNameMatch) = @_;
+
+    my $select = $isCount         ? 'COUNT(*)'   : join(',', @MRI_UPLOAD_FIELDS);
+    my $where  = $isBaseNameMatch ? 'LIKE ?' : '=?';
+
+    my $query = "SELECT $select "
+        .       "FROM mri_upload "
+        .       "JOIN tarchive USING(TarchiveID) "
+        .       "WHERE ArchiveLocation $where ";
+
+    try {
+        return $self->db->pselect(
+            $query,
+            $isBaseNameMatch ? ('%' . basename($tarchiveLocation) . '%') : $tarchiveLocation
+        );
+    } catch(NeuroDB::DatabaseException $e) {
+        NeuroDB::objectBroker::ObjectBrokerException->throw(
+            errorMessage => sprintf(
+                "Failed to retrieve mri upload records. Reason:\n%s",
```

Comment:
Great error message! 👍 

```
ERROR: Failed to retrieve mri upload records. Reason:
The following database commands failed:
	PREPARE s FROM 'SELECT COUNT(*) FROM  JOIN tarchive USING(TarchiveID) WHERE ArchiveLocation LIKE ? ';
	SET @x1='%DCM_2012-10-11_MTL0002_999145_NAPEN00_20121011_144212427.tar%';
	EXECUTE s USING @x1;
Error obtained:You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'JOIN tarchive USING(TarchiveID) WHERE ArchiveLocation LIKE '%DCM_2012-10-11_MTL0' at line 1 (error code 1064)
```

Reply:
...and you can copy this directly in your MySQL client for testing/debugging .  😄